front/components/agent_builder/types.ts

```diff
@@ -207,3 +215,64 @@ export interface ActionSpecification {
   cardIcon: NonNullable<React.ComponentProps<typeof Icon>["visual"]>;
   flag: WhitelistableFeature | null;
 }
+
+// MCP configuration types used by the agent builder.
+export type AssistantBuilderMCPServerConfiguration = {
+  mcpServerViewId: string;
+  dataSourceConfigurations: DataSourceViewSelectionConfigurations | null;
+  tablesConfigurations: DataSourceViewSelectionConfigurations | null;
+  childAgentId: string | null;
+  reasoningModel: ReasoningModelConfigurationType | null;
+  timeFrame: TimeFrame | null;
+  additionalConfiguration: AdditionalConfigurationType;
+  dustAppConfiguration: DustAppRunConfigurationType | null;
+  jsonSchema: JSONSchema | null;
+  _jsonSchemaString: string | null;
+  secretName: string | null;
+};
+
+export type AssistantBuilderMCPConfiguration = {
+  type: "MCP";
+  configuration: AssistantBuilderMCPServerConfiguration;
+  name: string;
+  description: string;
+  configurationRequired?: boolean;
+};
+
+export type AssistantBuilderMCPConfigurationWithId =
+  AssistantBuilderMCPConfiguration & {
+    id: string;
+  };
+
+export function getDefaultMCPServerActionConfiguration(
+  mcpServerView?: MCPServerViewType
+): AssistantBuilderMCPConfiguration {
```

Comment:
AgentBuilder

not flagging others but I think we probably want to rename every var from assistantbuilder to agentbuilder